phpseclib/Math/BigInteger.php

```diff
@@ -752,18 +750,17 @@ function __toString()
     /**
      * __clone() magic method
      *
-     * Although you can call BigInteger::__toString() directly in PHP5, you cannot call BigInteger::__clone() directly
-     * in PHP5.  You can in PHP4 since it's not a magic method, but in PHP5, you have to call it by using the PHP5
-     * only syntax of $y = clone $x.  As such, if you're trying to write an application that works on both PHP4 and
-     * PHP5, call BigInteger::copy(), instead.
-     *
      * @access public
-     * @see self::copy()
      * @return \phpseclib\Math\BigInteger
      */
     function __clone()
     {
-        return $this->copy();
+        $temp = new static();
+        $temp->value = $this->value;
+        $temp->is_negative = $this->is_negative;
+        $temp->precision = $this->precision;
+        $temp->bitmask = $this->bitmask;
+        return $temp;
     }
```

Comment:
> Altho that said, that example does make me think a `getPrecision` method is needed lol

Definitely! :)

Thanks for looking into this!

I think I didn't explain myself very well. I did not mean that one should stop using the `clone` keyword and go back to using the assignment operator ("="). **Instead, I meant that when one uses the `clone` keyword, one does not need to define a `__clone()` method.** Unless the cloned object contains other objects.

I tried following code:
```
class MyClass
{
    public $a;
    public $b;
}

$o = new MyClass;

$o->a = 'a';
$o->b = 'b';

$o2= clone $o;
$o2->a = 'c';

var_dump($o);
var_dump($o2);
```

Which produces following:
```
object(MyClass)#1 (2) {
  ["a"]=>
  string(1) "a"
  ["b"]=>
  string(1) "b"
}
object(MyClass)#2 (2) {
  ["a"]=>
  string(1) "c"
  ["b"]=>
  string(1) "b"
}
```

As can be seen, `MyClass` does not define a `__clone()` method. Nevertheless, the properties were truly copied - they are not references like in the case of an assignment. Your quote from the manual talks about **objects inside the cloned object**. So, if the $a and $b in my example were objects, then yes, I would need to define the `__clone()` method like you do in your code, and copy them manually. Because otherwise they would be references to the original object.

So, in your code, you treat `value`, `is_negative`, `precision` and `bitmask` as if they were **objects** and manually copy them. Which doesn't break anything, but is unnecessary - it is already done automatically before `__clone()` is called. The `__clone()` method here just burns cycles. :)

So, essentially, the `__clone()` method can be removed and the cloning in this case would work just as well.